### PR TITLE
[Feature] CX-3388 Announce camera alerts

### PR DESCRIPTION
**Pull request by josokinas**
# Problem
When camera alerts appears, currently screen reader does not inform user about this happening.

# Solution
In order to make this accessible:
- shutter button gets disabled when camera error happens (only when camera is not working but still accessible on timeout)
- component for displaying error messages (that is also used for camera alerts) can now have a role and option to focus on render (both controllable via props)

Interesting feature here is that screen reader on focus of an "alertdialog" keeps reading the whole content including, in our case, a button to close the modal. However this seems to be a behaviour, e.g. even in official W3 example this also happens: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [`n/a`] Have new automated tests been implemented?
- [`n/a`] Have new manual tests been written down?
- [x] Have tests passed locally?
- [`n/a`] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Photo/Selfie.js [hunk 1]
```diff
@@ -94,13 +99,14 @@ export default class Selfie extends Component<Props, State> {
...
       >
-        <Timeout seconds={ 10 } onTimeout={ this.handleTimeout } />
+        { !hasCameraError && <Timeout seconds={ 10 } onTimeout={ this.handleTimeout } /> }
         <ToggleFullScreen />
         <FaceOverlay />
         <div className={style.actions}>
           <button
             aria-label={translate('accessibility.shutter')}
             className={style.btn}
+            disabled={hasCameraError}
```
Comment (kulak-at) [inline src/components/Photo/Selfie.js]: Why don't you use the Button component here? Now you have to implement styles for disabled state in two places.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Photo/Selfie.js (hunk 1 above)
Comment (josokinas) [reply src/components/Photo/Selfie.js]: Shutter is not a standard button, it's an actual shutter:

Screenshot 2019-05-07 at 09 59 44

Also, I'm doing as less refactoring as possible as it's out of the scope of the ticket.

File: src/components/Error/index.js [hunk 2]
```diff
@@ -1,28 +1,51 @@
...
-  const errorType = error.type === 'error' ? 'error' : 'warning'
-  return (
-    <div className={classNames(style[`container-${errorType}`], className)}>
-      { withArrow && <div className={classNames(style.roundedTriangle, style[`${errorType}Triangle`])} /> }
-      <div className={style.title}>
-        <span className={style[`title-icon-${errorType}`]}/>
-        <span className={style['title-text']}>{renderMessage(translate(message))}</span>
+class Error extends Component {
+  componentDidMount() {
+    if (this.props.focusOnRender && this.container) {
```
Comment (stefaniacardenas) [inline src/components/Error/index.js]: Would this work by just using `autofocus=true` in the parent div?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Error/index.js (hunk 2 above)
Comment (josokinas) [reply src/components/Error/index.js]: No, regular `autoFocus` does not move the focus of tabbing order, that's why you need to hard-focus on the element.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Video/NotRecording.js [hunk 3]
```diff
@@ -8,21 +8,23 @@ import { localised } from '../../locales'
...
-    { !hasError && <Timeout key="notRecording" seconds={ 12 } onTimeout={ onTimeout } /> }
+    { !hasTimeoutError && !hasCameraError ? <Timeout key="notRecording" seconds={ 12 } onTimeout={ onTimeout } /> : null }
     <div className={style.actions}>
       <div className={classNames(style.captureActionsHint, style.recordAction)}>
         { translate('capture.liveness.press_record') }
       </div>
       <button
         aria-label={translate('accessibility.start_recording')}
         className={classNames(style.btn, style.startRecording)}
+        disabled={hasCameraError}
```
Comment (stefaniacardenas) [inline src/components/Video/NotRecording.js]: The recording button should not be clickable if recording took too long. Will this handle this case? I think the current behaviour does not allow you to click on the recording button if recording took too long

Comment (josokinas) [reply src/components/Video/NotRecording.js]: No, this only disables when camera is actually unavailable. Any reason for disabling in the state of a person taking too long to shoot? I don't see this as being a case, unless it's already recording a video?

Comment (josokinas) [reply src/components/Video/NotRecording.js]: @stefaniacardenas After chatting with @seewah I understand what is intended here, and made a fix. Now for videos:
- on "recording" component interaction is disabled when any error (timeout/camera) happens
- on "not recording" component interaction is disabled not only when camera error happens but also when recording is taking too long (hence it is still enabled when "camera became inactive" occurs, this is needed behaviour)

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Video/NotRecording.js (hunk 3 above)
Comment (stefaniacardenas) [reply src/components/Video/NotRecording.js]: Perfect!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-648-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://648-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**